pandas/tests/scalar/timedelta/test_constructors.py

```diff
@@ -348,47 +291,254 @@ def test_td_constructor_on_nanoseconds(constructed_td, conversion):
     assert constructed_td == Timedelta(conversion)
 
 
-def test_td_constructor_value_error():
-    msg = "Invalid type <class 'str'>. Must be int or float."
-    with pytest.raises(TypeError, match=msg):
-        Timedelta(nanoseconds="abc")
-
-
-def test_timedelta_constructor_identity():
-    # Test for #30543
-    expected = Timedelta(np.timedelta64(1, "s"))
-    result = Timedelta(expected)
-    assert result is expected
-
-
 @pytest.mark.parametrize(
-    "constructor, value, unit, expectation",
+    ("args", "kwargs"),
     [
-        (Timedelta, "10s", "ms", (ValueError, "unit must not be specified")),
-        (to_timedelta, "10s", "ms", (ValueError, "unit must not be specified")),
-        (to_timedelta, ["1", 2, 3], "s", (ValueError, "unit must not be specified")),
+        ((), {}),
+        (("ps",), {}),
+        (("ns",), {}),
+        (("ms",), {}),
+        ((), {"seconds": 3}),
+        (("ns",), {"minutes": 2}),
     ],
 )
-def test_string_with_unit(constructor, value, unit, expectation):
-    exp, match = expectation
-    with pytest.raises(exp, match=match):
-        _ = constructor(value, unit=unit)
+def test_other_args_ignored_if_timedelta_value_passed(args: tuple, kwargs: dict):
+    original = Timedelta(1)
+    new = Timedelta(original, *args, **kwargs)
+
+    assert new == original
+    if not any((args, kwargs)):
+        assert new is original
 
 
 @pytest.mark.parametrize(
     "value",
-    [
-        "".join(elements)
-        for repetition in (1, 2)
-        for elements in product("+-, ", repeat=repetition)
-    ],
+    (
+        None,
+        np.nan,
+        NaT,
+        pytest.param(
+            NA,
+            marks=pytest.mark.xfail(
+                reason="constructor fails",
+                raises=ValueError,
```

Comment:
Is this the desired behavior, or should it also return `NaT`?

Reply:
it is desired, yes.